src/Controls/src/Core/Page/Page.cs

```diff
@@ -323,10 +323,26 @@ public Task<string> DisplayActionSheetAsync(string title, string cancel, string
 			var args = new ActionSheetArguments(title, cancel, destruction, buttons);
 
 			args.FlowDirection = flowDirection;
+
+			// If page is no longer attached to a window (e.g., navigated away), ignore the action sheet request
+			if (Window is null)
+			{
+				// Complete the task with cancel result
+				args.SetResult(cancel);
```

Comment:
Could we add a log here? As a maui developer would be great to know why the Alert didn't show

Reply:
Not sure how that will be displayed. But yeah, the message looks good. I was thinking in something like `Trace.WriteLine` to be shown into the logs/output window